flink-connectors/flink-connector-kafka/src/main/java/org/apache/flink/streaming/connectors/kafka/FlinkKafkaProducer.java

```diff
@@ -647,8 +646,10 @@ private FlinkKafkaProducer(
                 new FlinkKafkaProducer.TransactionStateSerializer(),
                 new FlinkKafkaProducer.ContextStateSerializer());
 
+        checkArgument(
+                !org.apache.flink.util.StringUtils.isNullOrWhitespaceOnly(defaultTopic),
+                "defaultTopic cannot be null or empty string");
         this.defaultTopicId = checkNotNull(defaultTopic, "defaultTopic is null");
```

Comment:
As we have check the defaultTopicId is null or not, here we don't need to checknotNull again.
we can just assign this.defaultTopicId without any check.

this.defaultTopicId = defaultTopic